src/ui/App.tsx

```diff
@@ -118,43 +139,55 @@ function App(): JSX.Element {
     },
   };
 
-  useEffect(() => {
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    _signIn();
-  }, []);
-
-  const viewStyle = {flex: 1};
-
-  const cards = [];
-  const threads = threadListState.threads;
   // TODO: Once we take message fetching out of Card creation, only prerender
   // one card below the most recently swiped card. Until then, render more cards
   // and prevent rendering their messages to avoid getting stalled on slow
   // network.
-  let numCardsRendered = 10;
-  while (numCardsRendered-- && numCardsRendered < threads.length) {
-    const threadId = threads[numCardsRendered].id as string;
-    cards.push(
-      <Card
-        key={threadId}
-        threadId={threadId}
-        actions={threadActions}
-        cardSwipedAway={updateThreadListState}
-        preventRenderMessages={numCardsRendered > 2}
-      />,
-    );
-  }
+  const numCardsRendered = 10;
+  const cards = threadListState.threads
+    .slice(0, numCardsRendered)
+    .map((x, index) => {
+      const threadId = defined(x.id);
+      return (
+        <Card
+          key={threadId}
+          threadId={threadId}
+          actions={threadActions}
+          cardSwipedAway={updateThreadListState}
+          preventRenderMessages={index > 2}
+        />
+      );
+    });
+
+  // First card is at the bottom of the visual stack and last card is at the
+  // top. So reverse so we can have the first thread show up visibly at the top
+  // of the stack.
+  cards.reverse();
+
+  const style = StyleSheet.create({
+    view: cards.length
+      ? {flex: 1}
+      : {flex: 1, justifyContent: 'center', alignItems: 'center'},
+  });
 
   return (
     <React.Fragment>
-      <StatusBar barStyle="light-content" />
-      <SafeAreaView style={viewStyle}>
+      <SafeAreaView style={style.view}>
```

Comment:
Do both views need to have style.view here? Or can SafeAreaView just fill the viewport?

Reply:
They both needed the flex:1 IIRC, but I didn't test it recently.